lib/ulid/generator.rb

```diff
@@ -48,7 +48,18 @@ def octo_word(time = Time.now)
     end
 
     def time_48bit(time = Time.now)
-      time_ms = (time.to_f * 1000).to_i
+      # Avoid `time.to_f` since we want to accurately represent a whole number of milliseconds:
+      #
+      # > time = Time.new(2020, 1, 5, 7, 3, Rational(2, 1000))
+      # => 2020-01-05 07:03:00 +0000
+      # > (time.to_f * 1000).to_i
+      # => 1578207780001
+      #
+      # vs
+      #
+      # > time.to_i * 1000 + time.usec / 1000
+      # => 1578207780002
+      time_ms = time.to_i * 1000 + time.usec / 1000
```

Comment:
```suggestion
      # Avoid `time.to_f` since we want to accurately represent a whole number of milliseconds:
      #
      # > time = Time.new(2020, 1, 5, 7, 3, Rational(2, 1000))
      # => 2020-01-05 07:03:00 +0000
      # > (time.to_f * 1000).to_i
      # => 1578207780001
      #
      # vs
      #
      # > (time.to_r * 1000).to_i
      # => 1578207780002
      time_ms = (time.to_r * 1000).to_i
```

How about like this?

Reply:
Nice, thanks @kachick!